submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,109 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuLinkActive: 'menu__link--active',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroItemActive: 'hero__item--active',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+const sectionMenu = document.querySelector('.nav__menu');
+const sectionHero = document.querySelector('.hero');
+const url = './data/data.json';
+let selectMenuItem;
+let selectHeroItem;  
+
+async function fetchAsync () {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}  
+
+function createHtml(data) {
+    let navContent  = '';
+    let heroContent  = '';
+    data.pancakes.forEach((elem, index) => {
+        const menuItems = generateNavContent(elem, index); 
+        const heroItems = generateHeroContent(elem, index); 
+        navContent += menuItems;
+        heroContent += heroItems;
+    });
+    return {
+        navContent,
+        heroContent
+    };
+}
+
+function generateNavContent(elem, index) {
+    return `<li class=${styles.menuItem}>
+                <a class=${styles.menuLink} href='#' id=link${index+1}>${elem.title}</a>
+            </li>`;
+}
+
+function generateHeroContent(elem, index) {
+    return `<div class=${styles.heroItem} id='tab${index+1}'>
+                <div class=${styles.itemPrice}>Price for portion ${elem.price} $</div>
+                <figure class=${styles.heroImg} id='tab${index+1}'>
+                    <img class=${styles.itemIcon} src=${elem.path}>
+                    <figcaption class=${styles.itemTitle}>Pancakes ${elem.title}</figcaption>
+                </figure>
+                <div class=${styles.itemInfo}>${elem.description}</div>
+            </div>`;
+}
+
+function drawPage(details) {
+    sectionMenu.innerHTML = details.navContent;
+    sectionHero.innerHTML = details.heroContent;
+}
+
+function toggleTabs(){
+    sectionMenu.addEventListener('click', e => {
+        e.preventDefault();
```

Comment:
One more thing I noticed. You can handle the `click` event by doing this. And it will look better.
```suggestion
        if (e.target.matches('.menu__link')) {
           highlightMenuItem(e.target);
           highlightHeroItem(e.target);
        }
```
It helps you to remove `selectMenuItem` variable and maybe you'll notice how you can optimize the `highlightHeroItem` function.

Reply:
Hmm... is it really essential?  All my logic that's tied up in a click will break again... but okay, I'll try...

Reply:
Yes, it's an essential way to handle click events. Ping me if you'll need any help.

Reply:
I tried to figure out, but without result) So, I just need to remove e.prevent and put your proposal code (it is too easy) or I need to remove all from toggle func and put there you proposal code and then try to come up with some ideas?

Reply:
Yes, you need to remove `e.preventDefault();` and add this check. It will be enough.
In the `highlightMenuItem` you can remove `selectMenuItem` variable and just use `target`. It helps to simplify your code

Reply:
And let me know if this approach is clear for you. If you got it now, you can use it with confidence in the future

Reply:
Oh, and I started thinking in a very different direction, to remove all from func and start to think from what element I should found matched().

Reply:
No-no, it's already working fine. Just small optimization :)

Reply:
Sorry, you need to leave `e.preventDefault();` but place it inside this `if` statement. It's my fault to recommend you to remove it. I forgot that you use `a` tags in the menu.

Reply:
No problem! But I need more time, I cannot come up with an idea yet *(

Reply:
sorry guys, but I need help, I haven't any idea how to tackle this task with target and without extra variable...

Reply:
Maybe it'll help you
``` JavaScript
function highlightMenuItem(target) {
    const activeMenuLink = document.querySelector(styles.menuLinkActive);
    if (activeMenuLink) {
        activeMenuLink.classList.remove(styles.menuLinkActive);
    }
    target.classList.add(styles.menuLinkActive);
}
```

Reply:
Well, I was spinning around this solution, but I still didn't see anything... Thank you :)

Reply:
Thanks again! Everything is work, but with a slight difference ```const activeMenuLink = document.querySelector('.menu__link--active');``` I don't know why, but it didn't wanna work with variable...

Reply:
I think it doesn't work because maybe you missed '.' symbol in the selector. But it's no big deal
``` JavaScript
const activeMenuLink = document.querySelector(`.${menuLinkActive}`)
```

Reply:
Yep, you're right! Style property was taking from global obj styles, the path was style.menuLinkActive but, in querySelector method i put it without "." I guess it's too late and I'm sleepy)))

Reply:
Sure! Take some rest after this work.